### PR TITLE
Ti MSP432E4XX support

### PR DESCRIPTION
**Pull request by psanyi**

### CODE REVIEW

File: boards/arm/msp_exp432e401y_dk/msp_exp432e401y_dk_defconfig [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2021 pipei.sandor@gmail.com
+
+CONFIG_BOARD_MSP_EXP432E401Y_DK=y
+CONFIG_SOC_SERIES_MSP432E4XX=y
+CONFIG_SOC_MSP432E401Y=y
+CONFIG_CORTEX_M_SYSTICK=y
+
+# Floating point options
+CONFIG_FPU=y
```
Comment (ioannisg) [inline boards/arm/msp_exp432e401y_dk/msp_exp432e401y_dk_defconfig:10]: Any reason this is set on the _defconfig file? Are there any components directly making use of the FP?
Note that this by default enables FPU_SHARING as well.

Comment (psanyi) [reply boards/arm/msp_exp432e401y_dk/msp_exp432e401y_dk_defconfig:10]: I have created the board default config file based on 
msp_exp432p401r_launchxl_defconfig where the CONFIG_FPU is set to y.
https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/msp_exp432p401r_launchxl/msp_exp432p401r_launchxl_defconfig

For the moment I don't have any board sample projects which needs this option.

Comment (ioannisg) [reply boards/arm/msp_exp432e401y_dk/msp_exp432e401y_dk_defconfig:10]: then probably remove it? It is not an option that should be present in the board definition. It should be enabled by applications that do FP calculations, or subsystems/drivers that do it as well.

Comment (psanyi) [reply boards/arm/msp_exp432e401y_dk/msp_exp432e401y_dk_defconfig:10]: Ok. I will remove it.

**Comment by ioannisg** [comment]
Could you please, rebase your branch on upstream zephyr master, so only your commits show up here :)

**Comment by psanyi** [comment]
> Could you please, rebase your branch on upstream zephyr master, so only your commits show up here :)

Sure. My mistake. :)

File: boards/arm/msp_exp432e401y_dk/Kconfig [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# TI MSP-EXP432E401Y-DK configuration
+# Copyright (c) 2021 pipei.sandor@gmail.com
+
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_MSP_EXP432E401Y_DK
+
+endif # BOARD_MSP_EXP432E401Y_DK
```
Comment (MaureenHelm) [inline boards/arm/msp_exp432e401y_dk/Kconfig:8]: Remove this file, it doesn't do anything

File: boards/arm/msp_exp432e401y_dk/support/MSP432E401Y.ccxml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (MaureenHelm) [inline boards/arm/msp_exp432e401y_dk/support/MSP432E401Y.ccxml:1]: What is this file for?

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Thank you for the contribution!

Please add a board documentation page `boards/arm/msp_exp432e401y_dk/doc/index.rst`

File: boards/arm/msp_exp432e401y_dk/support/MSP432E401Y.ccxml (hunk 3 above)
Comment (psanyi) [reply boards/arm/msp_exp432e401y_dk/support/MSP432E401Y.ccxml:1]: The file is used by Code Composer Studio for XDS110 Debug Probe configuration (found on Ti SimpleLink Launchpads) .
Open OCD not using this file.

File: boards/arm/msp_exp432e401y_dk/Kconfig (hunk 2 above)
Comment (psanyi) [reply boards/arm/msp_exp432e401y_dk/Kconfig:8]: Ok. Wil be removed.

File: soc/arm/ti_simplelink/msp432e4xx/Kconfig.series [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# TI SimpleLink MSP432E4XX
+
+# Copyright (c) 2021, pipei.sandor@gmail.com
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_MSP432E4XX
+	bool "TI SimpleLink Family MSP432E4XX"
+	select ARM
```
Comment (ioannisg) [inline soc/arm/ti_simplelink/msp432e4xx/Kconfig.series:8]: @psanyi any reason you do not select CPU_HAS_ARM_MPU here?

File: boards/arm/msp_exp432e401y_dk/msp_exp432e401y_dk_defconfig [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2021 pipei.sandor@gmail.com
+
+CONFIG_BOARD_MSP_EXP432E401Y_DK=y
+CONFIG_SOC_SERIES_MSP432E4XX=y
+CONFIG_SOC_MSP432E401Y=y
+CONFIG_CORTEX_M_SYSTICK=y
```
Comment (ioannisg) [inline boards/arm/msp_exp432e401y_dk/msp_exp432e401y_dk_defconfig:7]: If MPU is supported, you should enable it here (CONFIG_ARM_MPU=y) and add also CONFIG_HW_STACK_PROTECTION=y

**Comment by psanyi** [comment]
> Thank you for the contribution!
> 
> Please add a board documentation page `boards/arm/msp_exp432e401y_dk/doc/index.rst`

Thanks for the review @MaureenHelm and @ioannisg .

I would like to add some gpio and uart support for this platform, to be able to run some basic sample applications like "blinky" and "hello". After than I will add the the documentation also.

File: boards/arm/msp_exp432e401y_dk/msp_exp432e401y_dk_defconfig (hunk 5 above)
Comment (psanyi) [reply boards/arm/msp_exp432e401y_dk/msp_exp432e401y_dk_defconfig:7]: Yes. I agree.

**Pull request closed without merging**